language/move-stdlib/src/natives/mod.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+pub mod bcs;
+pub mod event;
+pub mod hash;
+pub mod signer;
+pub mod vector;
+
+#[cfg(feature = "testing")]
+pub mod unit_test;
+
+#[cfg(feature = "testing")]
+pub mod debug;
+
+use move_core_types::{account_address::AccountAddress, identifier::Identifier};
+use move_vm_runtime::native_functions::NativeFunction;
+
+pub fn all_natives(
+    addr: AccountAddress,
+) -> Vec<(AccountAddress, Identifier, Identifier, NativeFunction)> {
+    const NATIVES: &[(&str, &str, NativeFunction)] = &[
+        ("BCS", "to_bytes", bcs::native_to_bytes),
+        (
+            "Event",
+            "write_to_event_store",
+            event::native_write_to_event_store,
+        ),
+        ("Hash", "sha2_256", hash::native_sha2_256),
+        ("Hash", "sha3_256", hash::native_sha3_256),
+        ("Signer", "borrow_address", signer::native_borrow_address),
+        ("Vector", "length", vector::native_length),
+        ("Vector", "empty", vector::native_empty),
+        ("Vector", "borrow", vector::native_borrow),
+        ("Vector", "borrow_mut", vector::native_borrow),
+        ("Vector", "push_back", vector::native_push_back),
+        ("Vector", "pop_back", vector::native_pop),
+        ("Vector", "destroy_empty", vector::native_destroy_empty),
+        ("Vector", "swap", vector::native_swap),
+        #[cfg(feature = "testing")]
+        ("Debug", "print", debug::native_print),
+        #[cfg(feature = "testing")]
+        (
+            "Debug",
+            "print_stack_trace",
+            debug::native_print_stack_trace,
+        ),
+        #[cfg(feature = "testing")]
+        (
+            "UnitTest",
+            "create_signers_for_testing",
+            unit_test::native_create_signers_for_testing,
+        ),
+    ];
+    NATIVES
+        .iter()
+        .cloned()
```

Comment:
I would have done this code exactly the same way. But is there a better way? Or will Rust optimize this enough that it is okay? Just feels weird to have to jump through this hoop to cleanly create the vector.

Reply:
I share your pain, but I can't think of another way that works. AFAIK the rust compiler is usually pretty smart on things like this so I'd trust it for now. (It's probably not gonna be a big deal even if it's not... The amount of data we're dealing with is limited.)